elements/lisk-p2p/src/peer_pool.ts

```diff
@@ -392,89 +389,36 @@ export class PeerPool extends EventEmitter {
 		if (this._nodeInfo) {
 			this._applyNodeInfoOnPeer(peer, this._nodeInfo);
 		}
-		peer.connect();
-
-		return peer;
-	}
-
-	public addDiscoveredPeer(
-		detailedPeerInfo: P2PDiscoveredPeerInfo,
-		inboundSocket?: SCServerSocket,
-	): Peer {
-		const peerConfig = {
-			connectTimeout: this._peerPoolConfig.connectTimeout,
-			ackTimeout: this._peerPoolConfig.ackTimeout,
-		};
-		const peer = new Peer(detailedPeerInfo, peerConfig, {
-			inbound: inboundSocket,
-		});
-		this._peerMap.set(peer.id, peer);
-		this._bindHandlersToPeer(peer);
-		if (this._nodeInfo) {
-			this._applyNodeInfoOnPeer(peer, this._nodeInfo);
-		}
-		peer.updatePeerInfo(detailedPeerInfo);
-		peer.connect();
 
 		return peer;
 	}
 
-	public addInboundPeer(
-		peerId: string,
-		peerInfo: P2PDiscoveredPeerInfo,
-		socket: SCServerSocket,
-	): boolean {
-		const existingPeer = this.getPeer(peerId);
-
-		if (existingPeer) {
-			// Update the peerInfo from the latest inbound socket.
-			existingPeer.updatePeerInfo(peerInfo);
-			if (existingPeer.state.inbound === ConnectionState.CLOSED) {
-				existingPeer.inboundSocket = socket;
-
-				return false;
-			}
-
-			return false;
-		}
-
-		this.addPeer(peerInfo, socket);
-
-		return true;
-	}
-
 	public addOutboundPeer(
 		peerId: string,
 		peerInfo: P2PDiscoveredPeerInfo,
-		socket: SCClientSocket,
-	): boolean {
+		socket?: SCClientSocket,
```

Comment:
do we need this `socket` option?

Reply:
Yes, we do as per discussion with @jondubois.

Reply:
@jondubois is it because we still keep the socket from discovery now?

Reply:
@shuse2 We need now, but yes we should aim to remove it when we change discovery.